tests/unit/s2n_random_test.c

```diff
@@ -894,6 +897,45 @@ int main(int argc, char **argv)
 {
     BEGIN_TEST_NO_INIT();
 
+    /* Feature probe */
+    {
+#if defined(OPENSSL_IS_AWSLC)
+        /* Negative test */
+    #if defined(S2N_LIBCRYPTO_SUPPORTS_ENGINE)
+        FAIL_MSG("Expected ENGINE feature probe to be disabled with AWS-LC");
+    #endif
+
+#else
+
+        /* Positive test
+         *
+         * We expect ENGINE APIs to be available if `OPENSSL_NO_ENGINE` is not
+         * defined.
+         */
+    #if !defined(OPENSSL_NO_ENGINE) && !defined(S2N_LIBCRYPTO_SUPPORTS_ENGINE)
```

Comment:
This should be positive.. "if is_openssl_1_0_2 we want the feature probe to be true".